### ISSUE TITLE
`@some_property.getter` is accepted and silently ignored

### ISSUE THREAD

**Issue by sterliakov** [microsoft/pyright#10633]
**Describe the bug**

In the code below, the actual getter type is `str`, while `pyright` thinks it's `int`. Overriding a property getter with explicit decorator should actually replace the getter type or emit a "not supported" error, but certainly not swallow it.

Furthermore, any method with arguments is also allowed.

**Code or Screenshots**

```python
class A:
    @property
    def foo(self) -> int:
        return 0
    
    @foo.getter
    def foo(self) -> str:
        return ''

reveal_type(A().foo)
```

```python
class A:
    @property
    def foo(self) -> int:
        return 0
    
    @foo.getter  # Typo, should've been @foo.setter, pyright - green
    def foo(self, v: int) -> str:
        return ''

reveal_type(A().foo)
```

or [playground](https://pyright-play.net/?strict=true&code=MYGwhgzhAECCBcAoaLoAEAOAnA9hgplgC4CeyqAJvgGbTU44AUE%2BI1AlNALQB80AlgDsiSVGOhZ8RAK5ZB0AAzkUy9PRwA6AOZSihVVVrrmrDtz4QiWUeJSSZc6AHIniRJIBu%2BMCAD6pAkZYRnYNdXZEIA)

Run results:

```
$ pyright a.py
/tmp/tmp.xTSMmFvPEV/a.py
  /tmp/tmp.xTSMmFvPEV/a.py:10:13 - information: Type of "A().foo" is "int"
0 errors, 0 warnings, 1 information
```

**VS Code extension or command-line**

CLI, `pyright 1.1.402`

**Comment by erictraut** [comment]
The `property` type requires significant special casing in static type checkers. It should have probably been defined as a generic type, but that's not how typeshed defines it. The special-case handling must therefore make a number of assumptions about how `property` is used. One of the assumptions that pyright (and other type checkers) currently make is that you won't redefine the `getter`. I've never seen this assumption violated in real-world code before, so I think this is a pretty reasonable assumption.

I'll note that no other type checkers (mypy, pyrefly, ty) include a check for this condition either.

I'm going to change this from a bug report to an enhancement request. For now, I'm going to reject the enhancement request and close the issue. I'm open to adding such a feature in the future if there's signal from enough other pyright users that it's something they'd like to see (e.g. in the form of upvotes).

**Comment by sterliakov** [comment]
Sorry for missing context. I discovered this while working on a similar `mypy` issue (https://github.com/python/mypy/pull/19313#discussion_r2160172088, link to the relevant comment with my research), and seems like this is a popular problem. This [GitHub code search](https://github.com/search?q=lang%3Apython+%2F%5E%5Cs%2B%40%5Cw%2B%5C.getter%2F&type=code&p=1) reveals multiple instances where `getter` is in fact misspelled `setter`, other cases are perverse enough to remain unsupported. 

Would it be possible to require that `@some_prop.getter` is at least applied to a zero-arg method? Ideally it should require same return type as the original defn, but this is less important IMO.
